packages/core/src/provisioning.ts

```diff
@@ -351,4 +378,154 @@ export class BasicProvisioner implements Provisioner {
 
         return prov;
     }
+
+    getAccountIdFromScimRequest(httpReq: IncomingMessage) {
+        const url = new URL(`http://localhost${httpReq.url || ""}`);
+        const secretToken = url.searchParams.get("token") || "";
+
+        // TODO: find secret token in the DB
+        if (secretToken === "asdrtyghj") {
+            return {
+                accountId: "472478c5-17e8-4ed5-8f51-05a9c5deaebb",
+                orgId: "e0bb91b4-2b35-4ba7-ba60-f4ac8470e7a3",
+            };
+        }
+
+        return {
+            accountId: null,
+            orgId: null,
+        };
+    }
+
+    validateScimUser(newUser: ScimUserRequestData): string | null {
+        if (!newUser.externalId) {
+            return "User must contain externalId";
+        }
+
+        if (!newUser.email) {
+            return "User must contain email";
+        }
+
+        if (!newUser.name.formatted) {
+            return "User must contain name.formatted";
+        }
+
+        if (newUser.meta.resourceType !== "User") {
+            return 'User meta.resourceType must be "User"';
+        }
+
+        return null;
+    }
+
+    async handleScimUsersPost(httpReq: IncomingMessage, httpRes: ServerResponse) {
+        let newUser: ScimUserRequestData;
+
+        const { accountId, orgId } = this.getAccountIdFromScimRequest(httpReq);
+
+        if (!accountId || !orgId) {
+            httpRes.statusCode = 400;
+            httpRes.end("Invalid SCIM Secret Token");
+            return;
+        }
+
+        try {
+            const body = await readBody(httpReq);
+            newUser = JSON.parse(body);
+        } catch (e) {
+            httpRes.statusCode = 400;
+            httpRes.end("Failed to read request body.");
+            return;
+        }
+
+        const validationError = this.validateScimUser(newUser);
+        if (validationError) {
+            httpRes.statusCode = 400;
+            httpRes.end(validationError);
+            return;
+        }
+
+        try {
+            const provisioning = await this.getProvisioning({ email: newUser.email, accountId });
```

Comment:
FYI: You don't need the `accountId` here, and in fact there is no guarantee that there even is one at this point, so just the email is fine.